e2e/pollers.go

```diff
@@ -230,3 +236,49 @@ func waitUntilPodDeleted(ctx context.Context, kube *kubeclient, podName string)
 		return err == nil, err
 	})
 }
+
+type Poller[T any] interface {
+	PollUntilDone(ctx context.Context, options *runtime.PollUntilDoneOptions) (T, error)
+}
+
+func pollVMSSOperation[T any](ctx context.Context, vmssName string, pollerOpts *runtime.PollUntilDoneOptions, vmssOperation func() (Poller[T], error)) (*T, error) {
+	var requestError azure.RequestError
+	var err error
+	var poller Poller[T]
+	for i := 0; i < vmssRetryAttempts; i++ {
```

Comment:
can we just use the `wait.PollImmediateWithContext` paradigm here instead like we do for all the others?

Reply:
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediateWithContext